editor-extender/spell-check/editor-spell-check-provider.ts

```diff
@@ -0,0 +1,224 @@
+import { ClassProvider, Injectable } from "@angular/core";
+import { ToolBarItem, EditorConfigProvider, EDITOR_CONFIG_TOKEN  } from "progress-sitefinity-adminapp-sdk/app/api/v1";
+import { HttpClient } from "@angular/common/http";
+import { Observable, forkJoin } from "rxjs";
+
+// This is webpack specific loader syntax for injecting css as <style> tag in header
+require("!style-loader!css-loader!./editor-spell-check-provider.css");
+
+/**
+ * Indicates the minimum value of certainty that is needed in order a correction to be applied.
+ */
+const MIN_CERTAINTY = 0.5;
+const HOST = 'https://api.cognitive.microsoft.com';
+const PATH = '/bing/v7.0/spellcheck';
+
+/**
+ * Mode of spellcheck
+ * Proof - Meant to provide Office Word like spelling corrections.
+ *         It can correct long queries, provide casing corrections and suppresses aggressive corrections.
+ * Spell - Meant to provide Search engine like spelling corrections.
+ *         It will correct small queries(up to length 65 tokens) without any casing changes and
+ *         will be more optimized (perf and relevance) towards search like queries.
+ */
+const PROOF_MODE = "proof";
+const SPELL_MODE = "spell";
+
+/**
+ * Currently the proof mode supports only these 3 cultures. For other cultures spell mode should be used.
+ * The list is subject to change.
+ * For more information check: https://docs.microsoft.com/bg-bg/azure/cognitive-services/bing-spell-check/proof-text
+ */
+const proofModeCultures = [
+    "en-US",
+    "es-ES",
+    "pt-BR"
+];
+
+/*
+ * NOTE: Replace this example key with your subscription key .
+ */
+const KEY = '';
```

Comment:
1. can we move this to the top, please as it is first thing we want developer to see, also put instructions how to obtain a key with url to azure ML portal.
2. Can we named more descriptive like: AZURE_LANGUAGE_SERVICES_API_KEY

Reply:
sure